### PR TITLE
Tests: SecureSockets: Enable SNI when connecting to IoT Core 

### PR DESCRIPTION
**Pull request by lundinc2**
Tests: SecureSockets: Enable SNI when connecting to IoT Core 

Description
-----------
When connecting to IoT Core, always try to use SNI to enable device's relying on Multi Account Registration.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/secure_sockets/test/iot_test_tcp.c [hunk 1]
```diff
@@ -2489,17 +2502,17 @@ static void prvServerDomainName( void )
     xResult = prvSetSockOptHelper( xSocket, xReceiveTimeOut, xSendTimeOut );
     TEST_ASSERT_EQUAL_INT32_MESSAGE( SOCKETS_ERROR_NONE, xResult, "Set sockopt Failed" );
 
+    /* Get the address struct for AWS Broker and SetSockOpt REQUIRE_TLS. */
+    xResult = prvAwsIotBrokerConnectHelper( xSocket, &xAwsBrokerAddress );
+    TEST_ASSERT_EQUAL_INT32_MESSAGE( SOCKETS_ERROR_NONE, xResult, "Failed setup AWS Broker address struct." );
+
```
Comment (AniruddhaKanhere) [inline libraries/abstractions/secure_sockets/test/iot_test_tcp.c:2508]: What is the purpose of moving this here? (from its previous location)

**Review by AniruddhaKanhere — COMMENTED** [review]
Please see the comment/question

File: libraries/abstractions/secure_sockets/test/iot_test_tcp.c (hunk 1 above)
Comment (lundinc2) [reply libraries/abstractions/secure_sockets/test/iot_test_tcp.c:2508]: This test case expects that the SNI of an invalid hostname will fail, but the hostname was getting overwritten in the helper function. By entering an invalid hostname after using the helper function, the test's condition holds, and the Socket is unable to connect

File: libraries/abstractions/pkcs11/test/iot_test_pkcs11.c [hunk 2]
```diff
@@ -409,7 +409,7 @@ static MultithreadTaskParams_t xGlobalTaskParams[ pkcs11testMULTI_THREAD_TASK_CO
 /*-----------------------------------------------------------*/
 /* Stack size of each task. This can be configured in iot_test_pkcs11_config.h. */
 #ifndef pkcs11testMULTI_TASK_STACK_SIZE
-    #define pkcs11testMULTI_TASK_STACK_SIZE    ( configMINIMAL_STACK_SIZE * 4 )
+    #define pkcs11testMULTI_TASK_STACK_SIZE    ( configMINIMAL_STACK_SIZE * 6 )
```
Comment (lundinc2) [inline libraries/abstractions/pkcs11/test/iot_test_pkcs11.c:412]: I will revert this, I pushed the wrong branch.

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**